tar_scm.py

```diff
@@ -92,62 +97,188 @@ def is_sslverify_enabled(kwargs):
     return 'sslverify' not in kwargs or kwargs['sslverify']
 
 
+def is_proxy_defined():
+    """Returns ``True`` if the ``proxy`` option has been enabled or
+    not been set (default enabled) ``False`` otherwise."""
+    if os.environ.get('http_proxy') is not None:
+        return True
+    else:
+        return False
+
+
+def define_global_scm_command(scm_type, sslverify):
+    """Defines the global scm command (with or without proxy)"""
+
+    global svntmpdir
+    global hgtmpdir
+    global global_scm_command
+    global_scm_command = ['unknown']
+
+    logging.debug('Using scm command: [' + scm_type + ']')
+    if sslverify:
+            logging.debug('SSL verification is: [ ON ]')
+    else:
+            logging.debug('SSL verification is: [ OFF ]')
+
+    # git should honor the http[s]_proxy variables, but we need to
+    # guarantee this, the variables do not work every time
+    if scm_type == 'git':
```

Comment:
Now that #132 is merged, we have exactly those subclasses I was wishing for.  So we can do this via a method in each SCM subclass.